Framework/Algorithms/src/ExtractSpectra.cpp

```diff
@@ -156,6 +172,81 @@ void ExtractSpectra::execHistogram() {
   }
 }
 
+/** Returns a pointer to a cropped X Histrogram to be used as the X Histogram
+ *  for each of the spectra in the new cropped workspace
+ *  @param workspace :: The workspace from which to get the cropped X Histogram
+ *  @return XHistogram :: Pointer to cropped X Histrogram from the workspace
+ */
+const Kernel::cow_ptr<Mantid::HistogramData::HistogramX>
+ExtractSpectra::getCroppedXHistogram(const API::MatrixWorkspace &workspace) {
+  const auto hist = workspace.histogram(0);
+  auto begin = m_minXIndex;
+  auto end = m_maxXIndex;
+  if (m_isHistogramData) {
+    end -= 1;
+  }
+
+  auto cropped(hist);
+  cropped.resize(end - begin);
+
+  auto xEnd = hist.xMode() == Histogram::XMode::Points ? end : end + 1;
```

Comment:
Again this is copying what was done before but it looks to me like there's very little difference. This is the function which is called to set `m_isHistogramData`

```
bool MatrixWorkspace::isHistogramDataByIndex(const std::size_t index) const {
  bool isHist = (x(index).size() != y(index).size());

  // TODOHIST temporary sanity check
  if (isHist) {
    if (getSpectrum(index).histogram().xMode() != HistogramData::Histogram::XMode::BinEdges) {
      throw std::logic_error("In MatrixWorkspace::isHistogramData(): "
                             "Histogram::Xmode is not BinEdges");
    }
  } else {
    if (getSpectrum(index).histogram().xMode() != HistogramData::Histogram::XMode::Points) {
      throw std::logic_error("In MatrixWorkspace::isHistogramData(): "
                             "Histogram::Xmode is not Points");
    }
  }
  return isHist;
}
```

Which I'd say is fairly confusingly written, but I think it would be fine to replace `Histogram::XMode::Points` with `!m_isHistogramData` if you'd like.

Reply:
Thanks for the explanation. I think to avoid us introducing any problems by accident, lets keep it as is for now